ha/core/support_bundle/ha_support_bundle.py

```diff
@@ -0,0 +1,156 @@
+        directory = os.path.dirname(_sb_tmp_src)
+        if not os.path.exists(directory):
+            os.makedirs(directory)
+        if destination is None:
+            destination = os.path.join(_sb_tmp_src,
+                os.path.basename(source))
+        try:
+            shutil.copy2(source, destination)
+        except FileNotFoundError as fe:
+            raise SupportBundleError(f"File not found : {fe}")
+
+    @staticmethod
+    def __generate_tar(bundle_id: str, target_path: str):
+        """Generate tar.gz file at given path."""
+        component = 'ha'
+        target_path = target_path if target_path is not None \
+            else _sb_default_path
+        target_path = os.path.join(target_path, component)
+        tar_name = bundle_id if bundle_id else _sb_tar_name
+        tar_file_name = os.path.join(target_path, tar_name + '.tar.gz')
+        if not os.path.exists(target_path):
+            os.makedirs(target_path)
+        with tarfile.open(tar_file_name, 'w:gz') as tar:
+            tar.add(_sb_tmp_src,
+                arcname=os.path.basename(_sb_tmp_src))
+
+    @staticmethod
+    def __clear_tmp_files():
+        """Clean temporary files created by the support bundle."""
+        shutil.rmtree(_sb_tmp_src)
+
+    @staticmethod
+    def parse_args():
+        """Parsing support bundle arguments."""
+        parser = argparse.ArgumentParser(description='''Bundle ha logs ''')
+        parser.add_argument('-b', dest='bundle_id', help='Unique bundle id')
+        parser.add_argument('-t', dest='path', help='Path to store the created bundle',
+            nargs='?', default="/var/seagate/cortx/support_bundle/")
+        parser.add_argument('-c', dest='cluster_conf',\
+            help="Cluster config file path for Support Bundle")
+        parser.add_argument('-s', '--services', dest='services', nargs='+',\
+            default='', help='List of services for Support Bundle')
+        parser.add_argument('-d', '--duration', default='P5D', dest='duration',
+            help="Duration - duration for which log should be captured, Default - P5D")
+        parser.add_argument('--size_limit', default='500MB', dest='size_limit',
+            help="Size Limit - Support Bundle size limit per node, Default - 500MB")
+        parser.add_argument('--binlogs', type=HASupportBundle.str2bool, default=False, dest='binlogs',
+            help="Include/Exclude Binary Logs, Default = False")
+        parser.add_argument('--coredumps', type=HASupportBundle.str2bool, default=False, dest='coredumps',
+            help="Include/Exclude Coredumps, Default = False")
+        parser.add_argument('--stacktrace', type=HASupportBundle.str2bool, default=False, dest='stacktrace',
+            help="Include/Exclude stacktrace, Default = False")
+        parser.add_argument('--modules', dest='modules',
+            help="list of components & services to generate support bundle.")
+        args=parser.parse_args()
+        return args
```

Comment:
[root@cortx-ha-headless-svc SB123TEST]# /opt/seagate/cortx/ha/bin/ha_support_bundle -h
usage: ha_support_bundle [-h] [-b BUNDLE_ID] [-t [PATH]] [-c CLUSTER_CONF]
                         [-s SERVICES [SERVICES ...]] [-d DURATION]
                         [--size_limit SIZE_LIMIT] [--binlogs BINLOGS]
                         [--coredumps COREDUMPS] [--stacktrace STACKTRACE]
                         [--modules MODULES]

Bundle ha logs

optional arguments:
  -h, --help            show this help message and exit
  -b BUNDLE_ID          Unique bundle id
  -t [PATH]             Path to store the created bundle
  -c CLUSTER_CONF       Cluster config file path for Support Bundle
  -s SERVICES [SERVICES ...], --services SERVICES [SERVICES ...]
                        List of services for Support Bundle
  -d DURATION, --duration DURATION
                        Duration - duration for which log should be captured,
                        Default - P5D
  --size_limit SIZE_LIMIT
                        Size Limit - Support Bundle size limit per node,
                        Default - 500MB
  --binlogs BINLOGS     Include/Exclude Binary Logs, Default = False
  --coredumps COREDUMPS
                        Include/Exclude Coredumps, Default = False
  --stacktrace STACKTRACE
                        Include/Exclude stacktrace, Default = False
  --modules MODULES     list of components & services to generate support
                        bundle.